hail/src/main/scala/is/hail/lir/X.scala

```diff
@@ -400,17 +382,20 @@ class Block {
 
   def append(x: StmtX): Unit = {
     assert(x.parent == null)
+    x.parent = this
     if (last == null) {
       first = x
       last = x
-    } else {
```

Comment:
Does a check like this need to be added to prepend as well?

Reply:
Oh, you're right, if a `ControlX` is prepended we need to drop all of the existing statements, though I'd be surprised if we ever do that. Will fix.